src/main/java/org/mockbukkit/mockbukkit/entity/ai/BrainMock.java

```diff
@@ -0,0 +1,61 @@
+package org.mockbukkit.mockbukkit.entity.ai;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Maps;
+import org.bukkit.entity.memory.MemoryKey;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+import java.util.Optional;
+
+@ApiStatus.Internal
+public class BrainMock
+{
+
+	private static final String MEMORY_KEY_CANNOT_BE_NULL = "Memory key cannot be null";
+
+	private final Map<String, Object> memories = Maps.newHashMap();
```

Comment:
Why do you store every key as a string and not memory keys. The constructor of memorykey is private, so there should only be one instance of each memory key with z namespace value

Reply:
In the beginning i was trying to make it agnostic to the type. But looking back it does not make sense. I'll update it, thank you